Nothing happens when click on "Login in with Browser" button for some users.

The bug appears only when the user has some problems with permissions, UAC or something

**To Reproduce**
1. Click `Login in with Browser` button

**Log or error message**
```
java.io.IOException: Cannot run program "rundll32": CreateProcess error=740, The requested operation requires elevation
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
	at java.base/java.lang.Runtime.exec(Runtime.java:594)
	at java.base/java.lang.Runtime.exec(Runtime.java:418)
	at java.base/java.lang.Runtime.exec(Runtime.java:315)
	at com.sun.javafx.application.HostServicesDelegate$StandaloneHostService.showDocument(HostServicesDelegate.java:141)
	at javafx.application.HostServices.showDocument(HostServices.java:115)
	at com.faforever.client.fx.PlatformService.showDocument(PlatformService.java:47)
	at com.faforever.client.login.OAuthValuesReceiver.lambda$receiveValues$2(OAuthValuesReceiver.java:79)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.io.IOException: CreateProcess error=740, The requested operation requires elevation
	at java.base/java.lang.ProcessImpl.create(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:494)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:159)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
	... 15 more
```

**Expected behavior**
Show the error message to the user

**Screenshots**
No need

**OS**
Windows

--- a/src/main/java/com/faforever/client/fx/PlatformService.java
+++ b/src/main/java/com/faforever/client/fx/PlatformService.java
@@ -43,9 +43,42 @@ public class PlatformService {
 
   /**
    * Opens the specified URI in a new browser window or tab.
+   * Note: The code is copied from {@link com.sun.javafx.application.HostServicesDelegate#showDocument(String)}
+   * The only fix is that all any exceptions are intercepted by our side, and we can tell the user what happened wrong.
    */
   public void showDocument(String url) {
-    hostServices.showDocument(url);
+    final String[] browsers = {
+        "xdg-open",
+        "google-chrome",
+        "firefox",
+        "opera",
+        "konqueror",
+        "mozilla"
+    };
+
+    String osName = System.getProperty("os.name");
+    try {
+      if (osName.startsWith("Mac OS")) {
+        Runtime.getRuntime().exec(
+            "open " + url);
+      } else if (osName.startsWith("Windows")) {
+        Runtime.getRuntime().exec(
+            "rundll32 url.dll,FileProtocolHandler " + url);
+      } else { //assume Unix or Linux
+        String browser = null;
+        for (String b : browsers) {
+          if (browser == null && Runtime.getRuntime().exec(
+              new String[]{"which", b}).getInputStream().read() != -1) {
+            Runtime.getRuntime().exec(new String[]{browser = b, url});
+          }
+        }
+        if (browser == null) {
+          throw new Exception("No web browser found");
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
   }
 
   /**

